gobblin-modules/gobblin-azkaban/src/main/java/org/apache/gobblin/service/modules/orchestration/AzkabanClient.java

```diff
@@ -306,6 +317,22 @@ public AzkabanClientStatus deleteProject(String projectName) throws AzkabanClien
     return runWithRetry(callable, AzkabanClientStatus.class);
   }
 
+  /**
+   * Checks if the project with specified name exists in Azkaban
+   */
+  public Boolean projectExists(String projectName) throws AzkabanClientException {
+    try {
+      fetchProjectFlows(projectName);
+      return true;
+    } catch (AzkabanClientException e) {
+      // Azkaban does not return a strongly typed error code, so we are checking the message
+      if (e.getCause().getMessage().contains("doesn't exist")) {
+        return false;
```

Comment:
Shall we print the error message anyway ? relying on a strict string matching here seems to be not reliable enough.

Reply:
This is an outcome of a poorly-designed REST API on Azkaban side. It does not have a direct way to check for project existence, and it does not return machine-readable error codes - https://azkaban.readthedocs.io/en/latest/ajaxApi.html#fetch-jobs-of-a-flow . This can be unreliable, if they change the messages. I'll add a feature request for them to have better error reporting in the API.

Since the error is an expected behavior here (when project does not exist), we shouldn't log it every time it happens - it will flood the logs when nothing bad has happened.